packages/sdk/src/types/internal/actor.ts

```diff
@@ -33,11 +33,12 @@ export class InternalActor implements InternalPatchable<ActorLike> {
         const behavior = (this.behavior && this.behavior.behaviorType === actionEvent.behaviorType)
             ? this.behavior : undefined;
         if (behavior && behavior._supportsAction(actionEvent.actionName)) {
-            behavior._performAction(actionEvent.actionName, actionEvent.actionState, actionEvent.userId);
+            behavior._performAction(actionEvent.actionName, actionEvent.actionState,
+                this.actor.context.user(actionEvent.userId));

```

Comment:
This logic should just be moved to the `public 'recv-perform-action'` method in the execution protocol, and the `ActionEvent` interface should be updated to contain a user instead of a user id.